src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -286,6 +286,120 @@ def _deploy_arm_template_core(cli_ctx, resource_group_name,
     return sdk_no_wait(no_wait, smc.deployments.create_or_update, resource_group_name, deployment_name, properties)
 
 
+def _remove_comments_from_json(template):
+    from jsmin import jsmin
+
+    minified = jsmin(template)
+    # Get rid of multi-line strings. Note, we are not sending it on the wire rather just extract parameters to prompt for values
+    result = re.sub(r'"[^"]*?\n[^"]*?(?<!\\)"', '"#Azure Cli#"', minified, re.DOTALL)
+    return shell_safe_json_parse(result, preserve_order=True)
+
+
+# pylint: disable=too-many-locals, too-many-statements, too-few-public-methods
+def _deploy_arm_template_unmodified(cli_ctx, resource_group_name, template_file=None,
+                                    template_uri=None, deployment_name=None, parameters=None,
+                                    mode=None, rollback_on_error=None, validate_only=False, no_wait=False):
+    DeploymentProperties, TemplateLink, OnErrorDeployment = get_sdk(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES,
+                                                                    'DeploymentProperties', 'TemplateLink',
+                                                                    'OnErrorDeployment', mod='models')
+    template_link = None
+    template_obj = None
+    on_error_deployment = None
+    template_content = None
+    if template_uri:
+        template_link = TemplateLink(uri=template_uri)
+        template_content = _urlretrieve(template_uri).decode('utf-8')
+        template_obj = _remove_comments_from_json(template_content)
+    else:
+        template_content = read_file_content(template_file)
+        template_obj = _remove_comments_from_json(template_content)
+
+    if rollback_on_error == '':
+        on_error_deployment = OnErrorDeployment(type='LastSuccessful')
+    elif rollback_on_error:
+        on_error_deployment = OnErrorDeployment(type='SpecificDeployment', deployment_name=rollback_on_error)
+
+    template_param_defs = template_obj.get('parameters', {})
+    template_obj['resources'] = template_obj.get('resources', [])
+    parameters = _process_parameters(template_param_defs, parameters) or {}
+    parameters = _get_missing_parameters(parameters, template_obj, _prompt_for_parameters)
+
+    parameters = json.loads(json.dumps(parameters))
+
+    properties = DeploymentProperties(template=template_content, template_link=template_link,
+                                      parameters=parameters, mode=mode, on_error_deployment=on_error_deployment)
+
+    smc = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES)
+
+    class JsonCTemplate(object):
+        def __init__(self, template_as_bytes):
+            self.template_as_bytes = template_as_bytes
```

Comment:
The name `template_as_bytes` is misleading.

I debugged `test_resource.DeploymentTest.test_group_deployment`, and saw its value is actually a `str`:

![image](https://user-images.githubusercontent.com/4003950/161533073-0dd62ba6-4609-46f6-a798-540899217878.png)